Dockerfile

```diff
@@ -103,13 +110,47 @@ RUN --mount=target=. --mount=target=/root/.cache,type=cache \
   xx-go build ${GOBUILDFLAGS} -ldflags "$(cat /tmp/.ldflags) -extldflags '-static'" -tags "osusergo netgo static_build seccomp ${BUILDKITD_TAGS}" -o /usr/bin/buildkitd ./cmd/buildkitd && \
   xx-verify ${VERIFYFLAGS} /usr/bin/buildkitd
 
-FROM scratch AS binaries-linux
+# build buildkitd binary (debug variant)
+FROM buildkit-base AS buildkitd-debug
+# BUILDKITD_TAGS defines additional Go build tags for compiling buildkitd
+ARG BUILDKITD_TAGS
+ARG TARGETPLATFORM
+ARG GOBUILDFLAGS
+ARG VERIFYFLAGS="--static"
+ARG CGO_ENABLED=0
+RUN --mount=target=. --mount=target=/root/.cache,type=cache \
+  --mount=target=/go/pkg/mod,type=cache \
+  --mount=source=/tmp/.ldflags,target=/tmp/.ldflags,from=buildkit-version \
+  xx-go build ${GOBUILDFLAGS} -gcflags="all=-N -l" -ldflags "$(cat /tmp/.ldflags) -extldflags '-static'" -tags "osusergo netgo static_build seccomp ${BUILDKITD_TAGS}" -o /usr/bin/buildkitd ./cmd/buildkitd && \
```

Comment:
We should try to avoid duplicate command and configure the difference with an arg. If it gets too long we can use a heredoc.

Reply:
I can probably do the same thing as what `-ldflags` does to simplify things.